### PR TITLE
e2e, multi-net policies: Add ingress allow all and deny all tests

### PR DESCRIPTION
**Pull request by oshoval**
**- What this PR does and why is it needed**
Add multi network policy ingress allow all and deny all e2e tests.

Add a test that shows a bug: egress deny-all affects ingress
(reverse case of #4156), and a test that shows the workaround for it:
add ingress wildcard.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by oshoval** [comment]
is it fine that the tests run on draft ?

File: test/e2e/multihoming_utils.go
Comment (oshoval) [inline test/e2e/multihoming_utils.go:286]: lets wait for https://github.com/ovn-org/ovn-kubernetes/pull/4156#discussion_r1495550950
it can be reused in this PR it seems

**Comment by oshoval** [comment]
rebased only

**Comment by oshoval** [comment]
rebased on top of https://github.com/ovn-org/ovn-kubernetes/pull/4156
in order to reuse the generic function (see https://github.com/ovn-org/ovn-kubernetes/pull/4156#discussion_r1495550950)

/hold
until 4156 is merged

**Comment by oshoval** [comment]
Added the test that shows the bug and the WA of the reverse scenario presented on #4156 
where an Egress deny-all affects ingress (and it shouldn't)
The WA is to add an Ingress allow-all wild card

**Comment by oshoval** [comment]
Rebased, now that #4156 is merged

Ready for review please
/cc @maiqueb @qinqon

File: test/e2e/multihoming.go
Comment (oshoval) [inline test/e2e/multihoming.go:33]: btw maybe this approach is better than 
https://github.com/ovn-org/ovn-kubernetes/pull/4156/files#diff-8a0101c1714fa2f18fcb47cc2f71b4a820bc6e88fb23f0fcb557abafbfb3b924R825

**Comment by oshoval** [comment]
added this small commit
https://github.com/ovn-org/ovn-kubernetes/pull/4153/commits/c4d38b8d342aa6761edd6767a638ab6a99322b3e

**Review by qinqon — APPROVED** [review]
/lgtm

File: test/e2e/multihoming.go
Comment (maiqueb) [inline test/e2e/multihoming.go:33]: why `specific` ? what's its purpose ? it's really not immediately clear.

Comment (oshoval) [reply test/e2e/multihoming.go:33]: it is not "allow all" / "deny all", i can change it to IPBlock but then also need to see if there are more usages and add one for them as well

(the resolution we care about the permission flag itself is just to know if it is allow / deny all / other)

File: test/e2e/multihoming.go
Comment (maiqueb) [inline test/e2e/multihoming.go:1028]: this is / else does not look good. Also, you're assuming "all" policies which are not deny all mean it'll be able to access the server. While (I think ...) that's true today, it doesn't mean that will be true in the future. 

Maybe consider splitting the table in two ? (one expecting successes, another expecting failures)

Also open to other alternatives.

File: test/e2e/multihoming.go
Comment (maiqueb) [reply test/e2e/multihoming.go:33]: so `specific` means something that is not allow or deny all ? 

That name is really not intuitive. 

`ipBlock` will not work long term, since we want to add support for namespace / pod selectors. 
 
maybe splitting the table into successful / blocked tables will allow you to remove the need for it. 

Wondering if we can do that without duplicating a lof of the test code inside the table function.

File: test/e2e/multihoming.go
Comment (oshoval) [reply test/e2e/multihoming.go:1028]: i can split it for sure, it will just cost code duplication
the other alternative, is to be reactive and only once the assumption you mentioned is broken to split it

whatever you prefer

File: test/e2e/multihoming.go
Comment (oshoval) [reply test/e2e/multihoming.go:33]: lets try to split

File: test/e2e/multihoming.go
Comment (oshoval) [reply test/e2e/multihoming.go:1028]: btw it is require to split to 3 (or to 2 but one of them will still have some conditions)

Comment (oshoval) [reply test/e2e/multihoming.go:1028]: what about something like 
https://github.com/ovn-org/ovn-kubernetes/pull/4156/files#diff-8a0101c1714fa2f18fcb47cc2f71b4a820bc6e88fb23f0fcb557abafbfb3b924R825

else we will need to split to 3 (deny all, allow all, and the ones that exists already today)
(or 2 maybe if two of the 2 can be combined, with some condition)

**Comment by oshoval** [comment]
rebased on top of https://github.com/ovn-org/ovn-kubernetes/pull/4243
addressed comments

**Comment by oshoval** [comment]
Rebased now that https://github.com/ovn-org/ovn-kubernetes/pull/4243 is merged

File: test/e2e/multihoming.go
Comment (maiqueb) [inline test/e2e/multihoming.go:1343]: shoudn't this be done only if you managed to setup the underlay  ?

Comment (oshoval) [reply test/e2e/multihoming.go:1343]: yes and no, 
lets say you created some bridges and then failed, so it will remove the ones that it created

since it was already the state before, i just copy pasted tbh due to the above reason

**Review by trozet — APPROVED** [review]

**Pull request merged**
